### PR TITLE
[New Rule] New BBR Rules - Part 2

### PR DESCRIPTION
**Pull request by w0rk3r**
## Issues

https://github.com/elastic/ia-trade-team/issues/158

## Summary

Adds 5 New BBR Rules:

* Binary Content Copy via Cmd.exe
* Unusual Process Extension
* Downloaded Shortcut Files (Moved from a "Development" DR to BBR)
* Mofcomp Activity
* Office Test Registry Persistence

Adjusted Mitre mappings on:

* Uncommon Registry Persistence Change
* Windows Account or Group Discovery

### CODE REVIEW

File: rules_building_block/defense_evasion_cmd_copy_binary_contents.toml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+timestamp_override = "event.ingested"
+type = "eql"
+building_block_type = "default"
+
+query = '''
+process where host.os.type == "windows" and event.type == "start" and
+  process.name : "cmd.exe" and
+    (process.args : "type" and process.args : ">" or
+     process.args : "type" and process.args : ">>" or
+     process.args : "copy" and process.args : "/b")
```
Comment (Aegrah) [inline rules_building_block/defense_evasion_cmd_copy_binary_contents.toml:33]:
```suggestion
process where host.os.type == "windows" and event.type == "start" and
  process.name : "cmd.exe" and (
    (process.args : "type" and process.args : ">") or
     (process.args : "type" and process.args : ">>") or
     (process.args : "copy" and process.args : "/b"))
```

File: rules_building_block/defense_evasion_unusual_process_extension.toml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+query = '''
+process where host.os.type == "windows" and event.type == "start" and
+  process.executable : "?*" and not
+  process.name : ("*.exe", "*.com", "*.scr", "*.tmp", "*.dat") and
+  not process.executable : 
+    (
+      "MemCompression",
+      "Registry",
+      "vmmem",
+      "vmmemWSL",
```
Comment (Aegrah) [inline rules_building_block/defense_evasion_unusual_process_extension.toml:37]: Are these above actual process.executables? Never seen a process executable without some sort of path before.

File: rules_building_block/defense_evasion_cmd_copy_binary_contents.toml (hunk 1 above)
Comment (Aegrah) [reply rules_building_block/defense_evasion_cmd_copy_binary_contents.toml:33]: Think these quotes should be added to ensure the logic always functions as intended. Feel free to dispute if that's not the case.

Comment (w0rk3r) [reply rules_building_block/defense_evasion_cmd_copy_binary_contents.toml:33]: Nice catch

File: rules_building_block/defense_evasion_unusual_process_extension.toml (hunk 2 above)
Comment (w0rk3r) [reply rules_building_block/defense_evasion_unusual_process_extension.toml:37]: Yup

**Review by Aegrah — APPROVED** [review]

File: rules_building_block/defense_evasion_cmd_copy_binary_contents.toml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+timestamp_override = "event.ingested"
+type = "eql"
+building_block_type = "default"
+
+query = '''
+process where host.os.type == "windows" and event.type == "start" and
+  process.name : "cmd.exe" and (
+    (process.args : "type" and process.args : ">") or
+     (process.args : "type" and process.args : ">>") or
+     (process.args : "copy" and process.args : "/b"))
```
Comment (brokensound77) [inline rules_building_block/defense_evasion_cmd_copy_binary_contents.toml:33]:
```suggestion
    (process.args : "type" and process.args : (">", ">>")) or
    (process.args : "copy" and process.args : "/b"))
```

File: rules_building_block/defense_evasion_unusual_process_extension.toml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+severity = "low"
+tags = ["Domain: Endpoint", "OS: Windows", "Use Case: Threat Detection", "Tactic: Defense Evasion", "Data Source: Elastic Defend", "Rule Type: BBR"]
+timestamp_override = "event.ingested"
+type = "eql"
+building_block_type = "default"
+
+query = '''
+process where host.os.type == "windows" and event.type == "start" and
+  process.executable : "?*" and not
+  process.name : ("*.exe", "*.com", "*.scr", "*.tmp", "*.dat") and
```
Comment (brokensound77) [inline rules_building_block/defense_evasion_unusual_process_extension.toml:31]:
```suggestion
  process.executable : "?*" and 
  not process.name : ("*.exe", "*.com", "*.scr", "*.tmp", "*.dat") and
```

File: rules_building_block/defense_evasion_unusual_process_extension.toml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  not process.executable :
```
Comment (brokensound77) [inline rules_building_block/defense_evasion_unusual_process_extension.toml:32]: Want to simplify this and avoid all exceptions and just look for MZ header and not standard extension? Sure, some will FP, but really its a BBR looking for _unusual_ ext. Could even just ignore all MS signed executables

File: rules_building_block/execution_downloaded_shortcut_files.toml [hunk 6]
```diff
@@ -1,37 +1,34 @@
...
-/* leaving in development pending `file.Ext.windows.zone_identifier` landing in ECS then endpoint */
-
-sequence by process.entity_id with maxspan=2s
-                                           /* file.extension added to endpoint fields for 7.10 */
-  [file where host.os.type == "windows" and event.type == "creation" and file.extension == "lnk"]
-                                           /* not sure yet how the update will capture ADS */
-  [file where host.os.type == "windows" and event.type == "creation" and file.extension == "lnk:Zone.Identifier" and
-     /* non-ECS field - may disqualify conversion */
-     file.Ext.windows.zone_identifier > 1]
+file where host.os.type == "windows" and event.type == "creation" and file.extension == "lnk" and file.Ext.windows.zone_identifier > 1
```
Comment (brokensound77) [inline rules_building_block/execution_downloaded_shortcut_files.toml:31]: w00t

**Review by brokensound77 — APPROVED** [review]
few suggestions then lgtm

**Pull request merged**